quickstart/dotnet-core/Program.cs

```diff
@@ -207,102 +208,97 @@ public static async Task ForceReconnectAsync()
         {
-            return BasicRetryAsync(() => Connection.GetDatabase());
-        }
+            await InitializeAsync();
 
-        public static Task<System.Net.EndPoint[]> GetEndPointsAsync()
-        {
-            return BasicRetryAsync(() => Connection.GetEndPoints());
-        }
+            // Perform cache operations using the cache object...
+            Console.WriteLine("Running... Press 'q' to quit.");
+            ConsoleKey inputKey = ConsoleKey.A; // Init value, it may be anything different to Q
 
-        public static Task<IServer> GetServerAsync(string host, int port)
-        {
-            return BasicRetryAsync(() => Connection.GetServer(host, port));
-        }
+            while (inputKey != ConsoleKey.Q)
+            {
+                for (int i = 0; i < 2; i++)
+                {
+                    Task.Run(() => RunRedisCommands());
+                    Thread.Sleep(500);
+                }
 
-        static async Task Main(string[] args)
-        {
-            await InitializeAsync();
+                Thread.Sleep(2000);
 
-            IDatabase cache = await GetDatabaseAsync();
+                if (Console.KeyAvailable)
+                {
+                    inputKey = Console.ReadKey().Key;
+                }
+            }
 
-            // Perform cache operations using the cache object...
+            await CloseConnectionAsync(_connection);
+        }
+
+        private static async Task RunRedisCommands()
+        {
 
             // Simple PING command
             string cacheCommand = "PING";
-            Console.WriteLine("\nCache command  : " + cacheCommand);
-            Console.WriteLine("Cache response : " + cache.Execute(cacheCommand).ToString());
+            Console.WriteLine($"\nCache command: " + cacheCommand);
+            RedisResult pingResult = await BasicRetryAsync(async (db) => await db.ExecuteAsync(cacheCommand));
+            Console.WriteLine($"Cache response: {pingResult}");
 
             // Simple get and put of integral data types into the cache
             cacheCommand = "GET Message";
-            Console.WriteLine("\nCache command  : " + cacheCommand + " or StringGet()");
-            Console.WriteLine("Cache response : " + cache.StringGet("Message").ToString());
+            Console.WriteLine($"\nCache command: {cacheCommand} or StringGet()");
+            RedisValue getMessageResult = await BasicRetryAsync(async (db) => await db.StringGetAsync("Message"));
+            Console.WriteLine($"Cache response: {getMessageResult}");
 
             cacheCommand = "SET Message \"Hello! The cache is working from a .NET Core console app!\"";
-            Console.WriteLine("\nCache command  : " + cacheCommand + " or StringSet()");
-            Console.WriteLine("Cache response : " + cache.StringSet("Message", "Hello! The cache is working from a .NET Core console app!").ToString());
+            Console.WriteLine($"\nCache command: {cacheCommand} or StringSet()");
```

Comment:
> or StringSet() [](http://example.com/codeflow?start=63&length=15)

These log lines need a cleanup. We're actually calling StringSetAsync("blah"), so we should say that